### ISSUE TITLE
Workflow Email Activity does not work

### ISSUE THREAD

**Issue by urbanit** [OrchardCMS/Orchard#5840]
There is a null exception when calling the Template_Smtp_Wrapper template. Is something that I am missing?

**Comment by sebastienros** [comment]
Just had it work on a brand new setup from 1.9.x.

Feel free to re-open if you can provide repro steps.

**Comment by jeffolmstead** [comment]
I believe my repo works here. If you have the Email triggered off AFTER a timer delay activity (one minute is fine) then you get this call stack:

```
2015-10-28 10:02:08,878 [40] Orchard.Exceptions.DefaultExceptionPolicy - EndlessMountainSolutions - An unexpected exception was caught
System.NullReferenceException: Object reference not set to an instance of an object.
   at Orchard.Mvc.ViewEngines.Razor.WebViewPage`1.get_DisplayHelperFactory() in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\Mvc\ViewEngines\Razor\WebViewPage.cs:line 70
   at Orchard.Mvc.ViewEngines.Razor.WebViewPage`1.InitHelpers() in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\Mvc\ViewEngines\Razor\WebViewPage.cs:line 168
   at System.Web.Mvc.RazorView.RenderView(ViewContext viewContext, TextWriter writer, Object instance)
   at Orchard.DisplayManagement.Descriptors.ShapeTemplateStrategy.ShapeTemplateBindingStrategy.RenderRazorViewToString(String path, DisplayContext context) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\Descriptors\ShapeTemplateStrategy\ShapeTemplateBindingStrategy.cs:line 174
   at Orchard.DisplayManagement.Descriptors.ShapeTemplateStrategy.ShapeTemplateBindingStrategy.Render(ShapeDescriptor shapeDescriptor, DisplayContext displayContext, HarvestShapeInfo harvestShapeInfo, HarvestShapeHit harvestShapeHit) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\Descriptors\ShapeTemplateStrategy\ShapeTemplateBindingStrategy.cs:line 157
   at Orchard.DisplayManagement.Descriptors.ShapeTemplateStrategy.ShapeTemplateBindingStrategy.<>c__DisplayClass20_7.<Discover>b__14(DisplayContext displayContext) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\Descriptors\ShapeTemplateStrategy\ShapeTemplateBindingStrategy.cs:line 133
   at Orchard.DisplayManagement.Descriptors.ShapeAlterationBuilder.<>c__DisplayClass7_1.<BoundAs>b__1(DisplayContext displayContext) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\Descriptors\ShapeAlterationBuilder.cs:line 56
   at Orchard.DisplayManagement.Implementation.DefaultDisplayManager.Process(ShapeBinding shapeBinding, IShape shape, DisplayContext context) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\Implementation\DefaultDisplayManager.cs:line 203
   at Orchard.DisplayManagement.Implementation.DefaultDisplayManager.Execute(DisplayContext context) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\Implementation\DefaultDisplayManager.cs:line 99
   at Orchard.DisplayManagement.Implementation.DisplayHelper.ShapeExecute(Object shape) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\Implementation\DisplayHelper.cs:line 86
   at Orchard.DisplayManagement.ShapeDisplay.Display(Object shape) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\DisplayManagement\ShapeDisplay.cs:line 39
   at Orchard.Email.Services.SmtpMessageChannel.Process(IDictionary`2 parameters)
   at Orchard.Messaging.Services.DefaultMessageService.Send(String type, IDictionary`2 parameters) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\Messaging\Services\DefaultMessageService.cs:line 20
   at Orchard.Events.DelegateHelper.<>c__DisplayClass15_0`3.<BuildAction>b__0(Object target, Object[] p) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\Events\DelegateHelper.cs:line 122
   at Orchard.Events.DefaultOrchardEventBus.TryInvokeMethod(IEventHandler eventHandler, Type interfaceType, String messageName, String interfaceName, String methodName, IDictionary`2 arguments, IEnumerable& returnValue) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\Events\DefaultOrchardEventBus.cs:line 84
   at Orchard.Events.DefaultOrchardEventBus.TryInvoke(IEventHandler eventHandler, String messageName, String interfaceName, String methodName, IDictionary`2 arguments, IEnumerable& returnValue) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\Events\DefaultOrchardEventBus.cs:line 70
   at Orchard.Events.DefaultOrchardEventBus.TryNotifyHandler(IEventHandler eventHandler, String messageName, String interfaceName, String methodName, IDictionary`2 eventData, IEnumerable& returnValue) in C:\Users\Jeffrey\Documents\Visual Studio 2013\Projects\EMS-Orchard\dev\src\Orchard\Events\DefaultOrchardEventBus.cs:line 53
```

Having debugged it, you end up with the crashing error being in Orchard.Mvc.ViewEngines.Razor.WebViewPage --> InitHelpers() --> 

```
_display = DisplayHelperFactory.CreateHelper(ViewContext, this);
```

Also, the WorkContext in the line above is coming in as Null so there seems to be a problem getting the ViewContext when driven from the task handler (i.e. because of the timer delay). Will let you know if I find more (can this be reopened?)

**Comment by jeffolmstead** [comment]
Update, causes the same issue if you bypass the time delay and use Queue (which likely processes via a background task also).

**Comment by jeffolmstead** [comment]
Final comment here, the actual "break" occurs in line 70 of WebViewPage.cs:

```
return _displayHelperFactory ?? (_displayHelperFactory = WorkContext.Resolve<IDisplayHelperFactory>());
```

Because, as mentioned above, the WorkContext is null, it cannot resolve the DisplayHelperFactories. 

If you follow the rabbit trail far enough, you go through  Orchard.WorkContextExtensions where you try to GetWorkContext which ends you up in Orchard.Environment.WorkContextAccessor where you are trying to get the WorkContext out of the HttpContext.Items and the workContextKey is just not found. 

In short, it seems background tasks which utilize WorkContext will be failing at this point (I am pretty much on the latest of the 1.9.2 branch for these tests).

**Comment by jeffolmstead** [comment]
Looks like it is being addressed in Issue #5983 so no need to reopen this one.
